state/src/impls/mod.rs

```diff
@@ -17,5 +17,8 @@
 mod shard_level;
 mod top_level;
 
+#[cfg(test)]
```

Comment:
Why?

Reply:
I thought if I tag `mod tests;` with `#[cfg(test)]`, the tests directory will be compiled only in test mode.

Reply:
I meant why do you split tests in the other module?
It makes the test cannot check the private status.

Reply:
I thought the source code was too long because of tests code and also the private status can be check by the test function.

Reply:
Abstract the module if you think it's too long.
It's not the proper way in rust to extract unit tests to another file.
https://doc.rust-lang.org/book/second-edition/ch11-03-test-organization.html